src/Http/HttpServiceProvider.php

```diff
@@ -25,5 +30,35 @@ public function register()
         $this->app->bind(Middleware\CheckCsrfToken::class, function ($app) {
             return new Middleware\CheckCsrfToken($app->make('flarum.http.csrfExemptPaths'));
         });
+
+        $this->app->singleton('flarum.http.slugDrivers', function () {
+            return [
+                Discussion::class => [
+                    'default' => IdWithTransliteratedSlugDriver::class
+                ],
+                User::class => [
+                    'default' => UsernameSlugDriver::class
+                ],
+            ];
+        });
+
+        $this->app->singleton('flarum.http.selectedSlugDrivers', function () {
+            $settings = $this->app->make(SettingsRepositoryInterface::class);
+
+            $compiledDrivers = [];
+
+            foreach ($this->app->make('flarum.http.slugDrivers') as $resourceClass => $resourceDrivers) {
+                $driverKey = $settings->get("slug_driver_$resourceClass", 'default');
+
+                $driverClass = array_key_exists($driverKey, $resourceDrivers) ? $resourceDrivers[$driverKey] : $resourceDrivers['default'];
```

Comment:
the ternary looks unnecessary here? `$driverKey` falls back to the provided default `default` anyway.

Reply:
There's probably a cleaner way to write this. The default fallback in $driverKey assignment is based on the settings: which driver is selected to be active? But the ternary ensures that that driver is registered: if someone enables nicknames, sets the driver to a driver provided by nicknames, but then uninstalls nicknames, we don't want it to load up a nonexistent driver.

The ternary can be replaced with an `Arr::get`

Reply:
Oh you're right yes, I didn't notice. My bad.